Linking 3.4.3 on unix is missing -ldl

When packaging/building 3.4.3 on unix derivates, using the proposed options, but with GPU rendering enabled, linking libosdGPU fails with:

```
[   38s] /usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /tmp/libosdGPU.so.3.4.3.P1M2ss.ltrans1.ltrans.o: in function `OpenSubdiv::internal::GLApi::loadFunction(char const*) [clone .lto_priv.0]':
[   38s] /home/abuild/rpmbuild/BUILD/OpenSubdiv-3_4_3/glLoader/glApi.cpp:3185: undefined reference to `dlsym'
[   38s] /usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /tmp/libosdGPU.so.3.4.3.P1M2ss.ltrans0.ltrans.o: in function `OpenSubdiv::internal::GLApi::glApiUnload()':
[   38s] /home/abuild/rpmbuild/BUILD/OpenSubdiv-3_4_3/glLoader/glApi.cpp:3162: undefined reference to `dlclose'
[   38s] /usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /tmp/libosdGPU.so.3.4.3.P1M2ss.ltrans0.ltrans.o: in function `OpenSubdiv::internal::GLApi::glApiLoad()':
[   38s] /home/abuild/rpmbuild/BUILD/OpenSubdiv-3_4_3/glLoader/glApi.cpp:3143: undefined reference to `dlopen'
[   38s] /usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /tmp/libosdGPU.so.3.4.3.P1M2ss.ltrans0.ltrans.o:/home/abuild/rpmbuild/BUILD/OpenSubdiv-3_4_3/glLoader/glApi.cpp:3149: undefined reference to `dlsym'
```

An ugly, but feasible workaround is:
```
sed -i 's/${PLATFORM_GPU_LIBRARIES}/${PLATFORM_GPU_LIBRARIES} ${CMAKE_DL_LIBS}/' opensubdiv/CMakeLists.txt
```
Full build with fix applied could be found [here](https://build.opensuse.org/package/show/home:frispete:blender/OpenSubdiv).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -468,7 +468,8 @@ if(GLEW_FOUND AND GLEW_INCLUDE_DIR)
         ${PROJECT_SOURCE_DIR}/glLoader)
     set(OPENGL_LOADER_LIBRARIES
         ${GLEW_LIBRARY}
-        ${OPENGL_gl_LIBRARY})
+        ${OPENGL_gl_LIBRARY}
+        ${CMAKE_DL_LIBS})
 
 elseif(OPENGL_FOUND)
 
@@ -477,7 +478,8 @@ elseif(OPENGL_FOUND)
         ${OPENGL_INCLUDE_DIR}
         ${PROJECT_SOURCE_DIR}/glLoader)
     set(OPENGL_LOADER_LIBRARIES
-        ${OPENGL_gl_LIBRARY})
+        ${OPENGL_gl_LIBRARY}
+        ${CMAKE_DL_LIBS})
 
 endif()
 

